backend/src/api/users/users.api.ts

```diff
@@ -61,11 +61,15 @@ const initUsersApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
     ) {
       const { id } = req.params;
 
-      const isDeleted = await userService.delete(Number(id));
+      const isDeleted = await userService.delete(req.user.id, Number(id));
+      const status =
+        isDeleted === undefined
+          ? HttpCode.BAD_REQUEST
+          : isDeleted
+          ? HttpCode.OK
+          : HttpCode.NOT_FOUND;
 
-      return rep
-        .status(isDeleted ? HttpCode.OK : HttpCode.NOT_FOUND)
-        .send(isDeleted);
+      return rep.status(status).send(isDeleted);
```

Comment:
Return prev implementation

Reply:
The only thing that the status should be `isDeleted ? HttpCode.NO_CONTENT : HttpCode.NOT_FOUND`

Reply:
ok, 400 status was said in qa trello card, that's why I implemented it

Reply:
maybe there is sense to add 400 status throwing an error in user service? for now, it will send a 404 in case when user is trying to delete himself, which is not completely correct, status 400 is more appropriate here I think

Reply:
Yes, let's do it in user service